// poke-ball/template/views/leafViews/list.js
import util from 'util';
import Mock from 'resource/mock';

export default {
  name: '{ModuleName}List',
  data () {
    return {
      formFilter: {
    <% config.list.forEach(function (item) { %>
      <% if (item.isFilter) { %>
        <%= item.name %>: '',
      <% } %>
    <% }) %>
      },
      {moduleName}s: {},
      pagination: {
        currentPage: 1,
        total: 0,
        pageSize: 10,
        pageSizes: [3, 5, 10, 20],
      }
    };
  },
  created () {
    // this.queryList();
  },
  methods: {

    handleSizeChange (size) {
      this.pagination.pageSize = size;
      this.pagination.currentPage = 1;
      this.queryList();
    },

    handleCurrentChange (index) {
      this.pagination.currentPage = index;
      this.queryList();
    },
    // 列表查询
    queryList () {
      Mock
        .postsResource()
        .get({
       <% config.list.forEach(function (item) { %>
        <% if (item.isFilter) { %>
          <%= item.name %>: this.formFilter.<%= item.name %>,
        <% } %>
       <% }) %>
          size: this.pagination.pageSize,
          page: (this.pagination.currentPage - 1) * this.pagination.pageSize
        })
        .then(response => {
          this.{moduleName}s = util.replaceEmptyWithHyphen(response.json());
          this.pagination.total = this.{moduleName}s.total;
        })
        .catch(error => { this.$notify({ message: '查询列表数据失败', type: 'error' }); });
    },

    handleSearch () {
      this.pagination.currentPage = 1;
      this.pagination.total = 0;
      this.queryList();
    },
  }
};
